File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for i in s:
            if not stack and ((i == ")" or i == "}") or i == "]"): return False
            if i == "(" or i == "{" or i == "[":
                stack.append(i)
            else:
                if i == "]" and stack.pop() != "[" :
                     return False
                elif i == "}" and stack.pop() != "{":
                     return False
                elif i == ")" and stack.pop() != "(":
                     return False
                    

                
        return not stack

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String brackets) {
         Stack<Character> para = new Stack<Character>();
      if(brackets.length() % 2 != 0){
          return false;
      }
      for(int i =0;i<brackets.length();i++){
         if(brackets.charAt(i) == '('){
            para.push('(');
         }else if(brackets.charAt(i) == '{'){
            para.push('{');
         }else if(brackets.charAt(i) == '['){
            para.push('[');
         }else if(brackets.charAt(i) == ')'){
            if(para.isEmpty()){
               return false;
            }else if(para.peek() != '('){
               return false;
            }else{
               para.pop();
            }
         }else if(brackets.charAt(i) == '}'){
            if(para.isEmpty()){
               return false;
            }else if(para.peek() != '{'){
               return false;
            }else{
               para.pop();
            }
         }else if(brackets.charAt(i) == ']'){
            if(para.isEmpty()){
               return false;
            }else if(para.peek() != '['){
               return false;
            }else{
               para.pop();
            }
         }
      }
      return para.isEmpty();

    }
}